source/server/admin/clusters_chunk_processor.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <string>
+
+#include "envoy/admin/v3/clusters.pb.h"
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/pure.h"
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/upstream/outlier_detection.h"
+#include "envoy/upstream/resource_manager.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/json/json_streamer.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Server {
+
+/**
+ * ClustersChunkProcessor is the interface for streaming Clusters data in a response
+ * for /clusters.
+ */
+class ClustersChunkProcessor {
+public:
+  virtual bool nextChunk(Buffer::Instance& response) PURE;
```

Comment:
I wasn't aware of how the Request was going to evolve. Since I am inheriting from this class here, I didn't want to cause problems by inheriting directly from Request for that reason. My interface only needs nextChunk so I went with this approach.

What are your thoughts given this perspective?

Reply:
I'm not sure Request is going to evolve further. If it does, and we deside to split out the clusters behavior from something else we could always split later. But if this is fundamentaly made per request maybe it'd be better to merge them now and split later if that ever seems needed.

Reply:
Sounds good to me. I'll make the adjustments.

Reply:
Now that we will be returning an `Http::Code::BadRequest` when the format received is invalid, I believe the current overall design makes sense. Let me explain.

The signature for the start member function on a `Request` looks like this:

`Http::Code start(Http::ResponseHeaderMap& response_headers) override;`

As you can see here, this member function naturally returns an `Http::Code`. My thought is that we can call `parse` on the ClustersParams and bubble back the `Http::Code` from that function invocation to the caller of `start`. Based on the format supplied by start, we can also instantiate the correct type of processor (similar to how we are doing that now). 

The signature for the `nextChunk` member function looks like this:

`bool nextChunk(Buffer::Instance& response) override;`

Here, we can delegate the request processing to the processor based on the format type. This is nice because we can employ composition instead of inheritance, which in this case, I believe is favorable since we probably don't want our inheritance hierarchy to get too deep. The request in this case, definitely has to handle parsing the request format query parameter but we have two separate processing strategies; we're employing the strategy pattern here since we have two different processing requirements. 

Interested in your thoughts given this insight.

Reply:
Hmm...it would appear that calling `Admin::makeStaticTextResponse` would be the member function to call to return a response early if there was a parsing failure on the format query parameter. Perhaps this can be done before the `Request` is even constructed. If that is the case, we could hypothetically just instantiate the correct type of request based on the result of parsing the format query parameter and have two different request types. 

I'll try this approach.